Remove unnecessary authenticator argument from EnableCustomSecurity annotation (#9)
sahabpardaz/spring-boot-rest-commons: doc/security.md

### Custom Authentication

> Again we are going to enable our custom security by registering it as a Spring bean:

Reviewer: ```suggestion
Again we are going to enable our custom security by adding the `@EnableCustomSecurity` annotation, but this time we register a bean as our custom `Authenticator` implementation:
```

```java
@SpringBootApplication
@EnableCustomSecurity(applicationBasePathPattern= "/protected-api/**", ignoredPaths = {"/ignored-auth/**"})
public class Application {
   @Bean
   public Authenticator authenticator() {
       return new CustomAuthenticator();
   }
   ...
}

```

As you see there are two arguments passed to the @EnableCustomSecurity:

- applicationBasePathPattern: specifies the base path pattern under which security (authentication/authorization) is
enabled. By default, we will enable security on all paths: "/**"
- ignoredPaths: excludes some paths to be ignored from base path. By default, no path is excluded. 

Besides these parameters, you can also pass another argument `authorityPrefix`. By default, Spring adds the prefix
"ROLE_" to the role names returned by the authentication object. But we do not like to manipulate the original role
names, so we do not append anythings to those names. Using this parameter, you can define your own prefix of choice.

> At the next step, we are going to implement the `CustomAuthenticator` class and register it as a Spring bean:

Reviewer: ```suggestion
At the next step, we are going to implement the `CustomAuthenticator` class:
```

In previous paragraph, you have mentioned the bean registration code.

```java

> @Component

Reviewer: ```suggestion
```

It does not need to be a component.
